test/support/factory.ex

```diff
@@ -472,7 +471,15 @@ defmodule Oli.Factory do
       type: :inline,
       result: :success,
       attempt_number: 1,
-      resource_access: insert(:resource_access)
+      resource_access: anonymous_build(:resource_access)
     }
   end
+
+  # HELPERS
+
+  defp anonymous_build(entity_name),
```

Comment:
Should we allow a map to be passed in, so specific attributes can be updated (like the original insert/build functions)?

Reply:
Definitely 💯